[BUG] __optnone__ conflicts with always_inline

**Describe the bug**


```
/private/var/tmp/usd/src/oiio-Release-2.1.16.0/src/include/OpenImageIO/benchmark.h:29:1: warning: 'always_inline' attribute ignored [-Wignored-attributes]
OIIO_FORCEINLINE T const& DoNotOptimize (T const &val);
^
/private/var/tmp/usd/src/oiio-Release-2.1.16.0/src/include/OpenImageIO/platform.h:306:52: note: expanded from macro 'OIIO_FORCEINLINE'
#    define OIIO_FORCEINLINE inline __attribute__((always_inline))
                                                   ^
/private/var/tmp/usd/src/oiio-Release-2.1.16.0/src/include/OpenImageIO/benchmark.h:430:33: note: conflicting attribute is here
inline T const & __attribute__((__optnone__))
```

I'm building 2.1.16.0, but the issue exists on top of tree OIIO, at slightly different line numbers.

**To Reproduce**
Steps to reproduce the behavior:

Build OpenImageIO with Xcode 13.4 toolset on a mac.

**Expected behavior**

Don't __optnone__ and always_inline at the same time.

--- a/src/include/OpenImageIO/benchmark.h
+++ b/src/include/OpenImageIO/benchmark.h
@@ -15,6 +15,15 @@
 #include <OpenImageIO/timer.h>
 
 
+#if (((OIIO_GNUC_VERSION && NDEBUG) || OIIO_CLANG_VERSION >= 30500 || OIIO_APPLE_CLANG_VERSION >= 70000 || defined(__INTEL_COMPILER)  || defined(__INTEL_LLVM_COMPILER)) \
+      && (defined(__x86_64__) || defined(__i386__))) \
+    || defined(_MSC_VER)
+#define OIIO_DONOTOPT_FORECINLINE OIIO_FORCEINLINE
+#else
+#define OIIO_DONOTOPT_FORECINLINE inline
+#endif
+
+
 OIIO_NAMESPACE_BEGIN
 
 /// DoNotOptimize(val) is a helper function for timing benchmarks that fools
@@ -23,10 +32,10 @@ OIIO_NAMESPACE_BEGIN
 /// May not work on all platforms. References:
 /// * Chandler Carruth's CppCon 2015 talk
 /// * Folly https://github.com/facebook/folly/blob/master/folly/Benchmark.h
-/// * Google Benchmark https://github.com/google/benchmark/blob/master/include/benchmark/benchmark_api.h
+/// * Google Benchmark https://github.com/google/benchmark/blob/main/include/benchmark/benchmark.h
 
 template <class T>
-OIIO_FORCEINLINE T const& DoNotOptimize (T const &val);
+OIIO_DONOTOPT_FORECINLINE T const& DoNotOptimize (T const &val);
 
 
 /// clobber_all_memory() is a helper function for timing benchmarks that

